File: LeetCode_3217/Main.java
public class Main {

}

//Definition for singly-linked list.

class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}


class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        ListNode result = null, resultHead = null;

        boolean[] toRemove = new boolean[100001];
        for (int num : nums) {
            toRemove[num] = true; // Mark values to remove
        }

        while (head != null) {
            if (!toRemove[head.val]) {
                if (resultHead == null) {
                    resultHead = head;
                    result = head;
                } else {
                    result.next = head;
                    result = result.next;
                }
            }
            head = head.next;
        }
        if (result != null) {
            result.next = null;
        }
        return resultHead;
    }
}

File: LeetCode_3217/Main.js

// Definition for singly-linked list.

function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? undefined : next)
}


/**
 * @param {number[]} nums
 * @param {ListNode} head
 * @return {ListNode}
 */
var modifiedList = function(nums, head) {
    const MAX = 100001;
    const cnt = new Array(MAX).fill(0);

    for (let num of nums) {
        cnt[num] += 1;
    }

    let result = null;
    let resultHead = null;

    while (head) {
        if (!cnt[head.val]) {
            if (resultHead === null) {
                result = head;
                resultHead = head;
            } else {
                result.next = head;
                result = result.next;
            }
        }
        head = head.next;
    }

    result.next = null; // ensure the ending of list
    return resultHead;
};